.github/workflows/ci_s2ntests.yml

```diff
@@ -0,0 +1,377 @@
+---
+name: s2nCodeBuildTesting
+
+on:
+  pull_request:
+    branches:
+      - master
+jobs:
+  integ:
+    needs: unit
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        openssl_version:
+          - openssl-1.0.2
+          - openssl-1.0.2-fips
+          - openssl-1.1.1
+          - libressl
+        gcc_version:
+          - 6
+          - 9
+      fail-fast: true
+    steps:
+      - uses: actions/setup-node@v1
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
+      - uses: hmarr/debug-action@master
+      - name: S2n Integ CodeBuild
+        uses: aws-actions/aws-codebuild-run-build@master
+        with:
+          project-name: 's2nGithubCodebuild'
+          env-vars-for-codebuild: |
+            BUILD_S2N,
+            CODECOV_IO_UPLOAD,
+            GCC_VERSION,
+            S2N_COVERAGE,
+            S2N_LIBCRYPTO,
+            TESTS,
+            requester,
+            event-name
+        env:
+          BUILD_S2N: true
+          CODECOV_IO_UPLOAD: true
+          S2N_LIBCRYPTO: ${{ matrix.openssl_version }}
+          S2N_COVERAGE: true
+          TESTS: "integration"
+          GCC_VERSION: ${{ matrix.gcc_version }}
+          requester: ${{ github.actor }}
+          event-name: ${{ github.event_name }}
+  integcorkedio:
+    needs: [unit, integ]
```

Comment:
Can we add comments explaining why we have these cross-job dependencies?

Also, there are a lot of interdependent jobs. Maybe what would work well is a dependency flow diagram at the top of the file?

Reply:
Sure- this is an attempt to slow down the workflow and prevent `rate exceeded` errors.

Reply:
Added a dep graph and some words in the CB Readme